### ISSUE TITLE
test: ensure test suite is passing on windows (wsl)

### ISSUE THREAD

**Issue by ssbarnea** [ansible/vscode-ansible#1451]
Currently we are not test `ui` and `e2e` on CI because:

- [ ] github hosted windows runners are not supporting WSL2,  docker and pomand (no nested virtualization
- [ ] executing `task ui` or `task e2e` is crashing windows runners with ambiguous message below

> The hosted runner: devtools-win-x64_04594ca6558b lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

We did switch from default windows-2022 (8GB RAM) to enlarged 16GB one but without any visible change.

Next step would to setup a permanent self-hosted windows based github runner on AWS and use this one for testing. This should also allow us to connect to the machine directly and see why it was crashing.

**Comment by alisonlhart** [comment]
If I'm reading [this](https://repost.aws/questions/QUg4UjAGV0RBaUza48MYKQlQ/how-to-enable-the-virtual-machine-platform-windows-feature-in-ec2-instance) correctly, we can't run WSL2 on an EC2 Windows instance because it doesn't support nested virtualization, same issue that used to be present in the GH runners.
I'm seeing that issue right now when I try to configure the new instance to use WSL 2. (edited) 

We may have to wait until the actions/runner-images folks get to my [request](https://github.com/actions/runner-images/issues/10563) for enabling WSL 2 before we can complete the WSL extension tests [issue](https://github.com/ansible/vscode-ansible/issues/1451).

**Comment by alisonlhart** [comment]
I don't think we have any paths forward until WSL 2 is enabled in github hosted runners.

**Comment by alisonlhart** [comment]
Using the Azure instance instead of AWS to do this. Description has been updated.

**Comment by alisonlhart** [comment]
`shell: bash` will not work for windows runners using WSL. Continuing to research and try other options. 
See: https://github.com/actions/runner/issues/1328

**Comment by vedantmgoyal9** [comment]
You can use https://github.com/vedantmgoyal9/setup-wsl2.

**Comment by gundalow** [comment]
For some projects under github.com/ansible and github.com/ansible-collections we use Azure Pipelines for testing against [different Windows versions](https://github.com/ansible-collections/ansible.windows/blob/894cf1d178721651a037ef67e902b998d82acdde/.azure-pipelines/azure-pipelines.yml#L132-L155), this integrates well with GitHub's CI reporting

We could have a specific Azure Pipelines jobs that only runs windows.

We have a decent amount of Azure Pipeline capacity for running jobs.
